libtrap/src/ifc_tls.c

```diff
@@ -1022,18 +1015,24 @@ static int client_socket_connect(tls_receiver_private_t *c, struct timeval *tv)
       }
       break;
    }
+   /* there was no successfull connection for whole servinfo struct */
+   if (p == NULL) {
+      VERBOSE(CL_VERBOSE_LIBRARY, "recv client: Connection failed.");
+      rv = TRAP_E_TIMEOUT;
+   }
+
+   /* catching all possible errors from setting up socket before atempting tls handshake */
+   if (rv != TRAP_E_OK) {
+      CHECK_AND_FREE(servinfo, freeaddrinfo);
+      return rv;
+   }
 
    if (p != NULL) {
       inet_ntop(p->ai_family, get_in_addr((struct sockaddr *)p->ai_addr), s, sizeof s);
```

Comment:
This PR adds check for the return value of `inet_ntop()` in ifc_tcpip.c -> It should be the same here.